pkg/cluster/update.go

```diff
@@ -213,7 +253,17 @@ func (u *simpleUpgrader) updateInPlace(ctx context.Context, cs *api.OpenShiftMan
 		return err
 	}
 
+	updateTracker, err := u.readBlob()
+	if err != nil {
+		return err
+	}
+
 	for _, vm := range sorted {
+		if isUpdated := vmIsUpdated(vm, role, updateTracker); isUpdated {
```

Comment:
just `is vmIsUpdated(vm, role, updateTracker) {`

Reply:
I feel like ideally you should either do the `if updated ... continue` approach in both updateInPlace and updatePlusOne, or do the filterOldVMs approach in both.  For testability, perhaps the latter although I kinda prefer the former.